### PR TITLE
Fix: Athena Sync version mapping

### PR DESCRIPTION
**Pull request by jfromme**
PR modifies the `DatasetDownload` results returned by the Athena client in two ways in the`/metrics/dataset/athena/download/sync` back end.

1. Removes `DatasetDownloads` with a datasetId of zero. These appear because of a `/0/` S3 key prefix in our publish buckets used for testing but which do not correspond to datasets in Postgres and so should not be counted in the metrics.
2.The Athena query is no longer able to fill in the version of the dataset being downloaded. Instead all `DatasetDownloads` coming from the Athena client have version set to zero. So the second change is to look up the most recent version of the dataset and replace the 0 version with the that version.

Also adds a test to verify these two changes.

### CODE REVIEW

**Review by jwagenaar — APPROVED** [review]
thanks John!

File: server/src/test/scala/com/pennsieve/discover/handlers/SyncHandlerSpec.scala [hunk 1]
```diff
@@ -140,5 +132,68 @@ class SyncHandlerSpec
...
+            version = ds1V2.version,
+            downloadedAt = convertToLocalOffset(req2.downloadedAt)
+          )
+        )
+    }
+  }
+
+  // I believe the DB connection returns the downloadedAt values in DatasetDownloads in the local time offset.
+  // Without the conversion to the local offset tests were failing locally on developer machines,
+  // but passing in CI where the system timezone is UTC.
```
Comment (rohanshah) [inline server/src/test/scala/com/pennsieve/discover/handlers/SyncHandlerSpec.scala:190]: fwiw another thing that could be happening, is that Postgres stores timestamps in microseconds, but the the java time classes like OffsetDateTime will use the system's precision, which can either be nanoseconds or microseconds. So when its in nanoseconds equality comparison fails in unit tests. And its also why it tends to fail locally (esp on newer macbooks) and pass in CI. Or at least that's what I've been seeing recently.

Comment (jfromme) [reply server/src/test/scala/com/pennsieve/discover/handlers/SyncHandlerSpec.scala:190]: When I run the tests on my laptop without using the `convertToLocalOffset()` function the results differ exactly by the offset that would be in effect on that date. At least by the time they are `OffsetDateTime` instances.

```scala
Expected :List(DatasetDownload(1,1,Some(AWSRequesterPayer),Some(REQID1),2020-11-10T10:03:01Z), DatasetDownload(1,2,Some(AWSRequesterPayer),Some(REQID2),2020-11-10T10:03:06Z))
Actual   :List(DatasetDownload(1,1,Some(AWSRequesterPayer),Some(REQID1),2020-11-10T05:03:01-05:00), DatasetDownload(1,2,Some(AWSRequesterPayer),Some(REQID2),2020-11-10T05:03:06-05:00))
```

Comment (rohanshah) [reply server/src/test/scala/com/pennsieve/discover/handlers/SyncHandlerSpec.scala:190]: ah gotcha! good to know!

**Pull request merged**